src/main/java/network/brightspots/rcv/TabulatorSession.java

```diff
@@ -54,15 +55,17 @@ class TabulatorSession {
     timestampString = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss").format(new Date());
   }
 
-  // validation will catch a mismatch and abort anyway, but let's log helpful errors for the CLI
-  // here also
-  private static void checkConfigVersionMatchesApp(ContestConfig config) {
+  // validation will catch a mismatch and abort anyway, but add another safety check just in case
+  private static void checkConfigVersionMatchesApp(ContestConfig config)
+      throws ConfigVersionIsNewerThanAppVersionException {
     String version = config.getRawConfig().tabulatorVersion;
 
     if (!version.equals(ContestConfig.AUTOMATED_TEST_VERSION)) {
-      //noinspection StatementWithEmptyBody
       if (ContestConfigMigration.isConfigVersionNewerThanAppVersion(version)) {
-        // It will log a severe message already, so no need to add one here.
+        // In practice, this should never be thrown because we should have checked the version
+        // while loading the config, and the version number can never change after a load+migration.
+        // Still, as a safety check, we'll throw here.
```

Comment:
Ah, I see -- the CLI cannot migrate the config, but the GUI does, because the GUI gives you an immediate representation of what it looks like migrated. This comment makes more sense now.